scripts/eqtl_hail_batch/generate_eqtl_spearman.py

```diff
@@ -416,20 +439,27 @@ def spearman_correlation(df):
     spearman_df['round'] = '1'
     # add celltype id
     celltype_id = celltype.lower()
+    print(f'celltype ID = {celltype_id}')
     spearman_df['cell_type_id'] = celltype_id
     # chromosome must be turned into a number solely
     # note, 'chr' + chromosome number was necessary in the previous step, 
     # as 'chr' indicates GrCh38 format
-    spearman_df.chrom = spearman_df.chrom.str.split('chr', expand=True)[1]
+    spearman_df['chrom'] = spearman_df.chrom.str.split('chr', expand=True)[1]
+    print(f'Spearman df chrom is {spearman_df.chrom}')
     # add association ID annotation after adding in alleles, a1, and a2
-    spearman_df.association_id = spearman_df.apply(lambda x: ':'.join(x[['chrom', 'bp', 'a1', 'a2', 'gene_symbol', 'cell_type_id', 'round']]), axis=1)
-    spearman_df.variant_id = spearman_df.apply(lambda x: ':'.join(x[['chrom', 'bp', 'a2']]), axis=1)
-    spearman_df.snp_id = spearman_df.apply(lambda x: ':'.join(x[['chrom', 'bp', 'a1', 'a2']]), axis=1)
+    spearman_df['association_id'] = spearman_df.apply(lambda x: ':'.join(x[['chrom', 'bp', 'a1', 'a2', 'gene_symbol', 'cell_type_id', 'round']]), axis=1)
+    spearman_df['variant_id'] = spearman_df.apply(lambda x: ':'.join(x[['chrom', 'bp', 'a2']]), axis=1)
+    spearman_df['snp_id'] = spearman_df.apply(lambda x: ':'.join(x[['chrom', 'bp', 'a1', 'a2']]), axis=1)
+    
     return spearman_df
 
 
 def merge_df_and_convert_to_string(*df_list):
     """Merge all Spearman dfs and convert to string using .to_string() on df"""
```

Comment:
Can you include more information on what qvalue is doing in this function, maybe even update the function name?

Reply:
added in a comment to explain this a bit more